pygatt/backends/bgapi/bgapi.py

```diff
@@ -297,30 +321,46 @@ def connect(self, address, timeout=5,
                 address_bytes, addr_type, interval_min, interval_max,
                 supervision_timeout, latency))
 
-        self.expect(ResponsePacketType.gap_connect_direct)
         try:
-            _, packet = self.expect(EventPacketType.connection_status,
-                                    timeout=timeout)
-            # TODO what do we do if the status isn't 'connected'? Retry? Raise
-            # an exception? Should also check the address matches the expected
-            # TODO i'm finding that when reconnecting to the same MAC, we geta
-            # conneciotn status of "disconnected" but that is picked up here as
-            # "connected", then we don't get anything else.
-            if self._connection_status_flag(
-                    packet['flags'],
-                    constants.connection_status_flag['connected']):
-                device = BGAPIBLEDevice(bgapi_address_to_hex(packet['address']),
-                                        packet['connection_handle'],
-                                        self)
+            self.expect(ResponsePacketType.gap_connect_direct)
+        except ExpectedResponseTimeout:
+            # If the connection doesn't occur because the device isn't there
+            # then you should manually stop the command.
+            self.send_command(CommandBuilder.gap_end_procedure())
+            self.expect(ResponsePacketType.gap_end_procedure)
+        else:
+            try:
+                _, packet = self.expect(EventPacketType.connection_status,
+                                        timeout=timeout)
+                # TODO what do we do if the status isn't 'connected'?
```

Comment:
I dunno. I looked at those and if they were my comments I would have probably removed them. But, I didn't feel like tearing out someone else's work without knowing what I was doing. Technically, they seem to refer to a different failure mode of getting a connection status packet back but it saying that the connection is not connected. That is not the same as the cases that I fixed. (But it is very close.) I would have to do more reading and testing to prove that the cases are the same. There might be a totally legit additional error handling case that this TODO refers to.